google/cloud/testing_util/example_driver.cc

```diff
@@ -61,17 +61,21 @@ int Example::Run(int argc, char const* const argv[]) try {
 
   std::string const command_name = argv[1];
   auto command = commands_.find(command_name);
-  if (commands_.end() == command) {
-    PrintUsage(argv[0], "Unknown command: " + command_name);
-    return 1;
-  }
+    if (commands_.end() == command) {
+        if (command_name == "--help") {
+            PrintUsage(argv[0], "");
+            return 0;
+        }
+        PrintUsage(argv[0], "Unknown command: " + command_name);
+        return 1;
+    }
 
   command->second({argv + 2, argv + argc});
 
   return 0;
 } catch (Usage const& u) {
   PrintUsage(argv[0], u.what());
-  return 1;
+  return 0;
```

Comment:
Perhaps this was bad advice.  It causes `//google/cloud/testing_util:example_driver_test` to fail, at least.

I'm not sure where that leaves us though, because I imagine that we want `samples command foo` to fail (when "foo" is a bad argument), but #6819 seems to want `samples command --help` to succeed, in addition to `samples --help`.

For example, with this PR applied we see ...
```
$ bazel-bin/google/cloud/iam/samples/iam_samples --help
Usage: iam_samples <command> [arguments]

Commands:
    :
$ echo $?
0
```
Good.  But ...
```
$ bazel-bin/google/cloud/iam/samples/iam_samples list-service-accounts
list-service-accounts <project-id>
Usage: iam_samples <command> [arguments]

Commands:
    :
$ echo $?
0
```
should probably fail, while ...
```
$ bazel-bin/google/cloud/iam/samples/iam_samples list-service-accounts --help
list-service-accounts <project-id>
Usage: iam_samples <command> [arguments]

Commands:
    :
$ echo $?
0
```
should probably succeed (if we agree with the whole premise).  But they both go through this single `return` because they both throw a `Usage`.

[Aside: The output in those last two "list-service-accounts" commands isn't the greatest either.]

Anyway, maybe @dbolduc, as the originator of #6819, can tell us more about how this should behave.